pkg/v1/remote/transport/retry.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transport
+
+import (
+	"log"
+	"net/http"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// Sleep for: 0.1, 0.3, 0.9, 2.7 seconds
+var DefaultBackoff = wait.Backoff{
+	Duration: 100 * time.Millisecond,
+	Factor:   3.0,
+	Jitter:   0.1,
+	Steps:    5,
+}
+
+var _ http.RoundTripper = (*RetryTransport)(nil)
+
+// RetryTransport wraps a RoundTripper and retries temporary network errors.
+type RetryTransport struct {
+	Inner   http.RoundTripper
+	Backoff wait.Backoff
+}
+
+// This is implemented by several errors in the net package.
+type temporary interface {
+	Temporary() bool
+}
+
+func (t *RetryTransport) RoundTrip(in *http.Request) (out *http.Response, err error) {
+	RetryTemporary(func() error {
+		out, err = t.Inner.RoundTrip(in)
+		return err
+	}, t.Backoff)
+	return
+}
+
+func RetryTemporary(f func() error, backoff wait.Backoff) (err error) {
```

Comment:
Should we check if `f` is `nil` here, and return an error? Otherwise, we'll panic on line 56.